File: backend/src/domain/value-objects/user-game-id.ts
export class UserGameId {
  private readonly _value: string;

  private constructor(value: string) {
    this._value = value;
  }

  static create(value: string): UserGameId {
    if (!value || value.trim() === '') {
      throw new Error('UserGame ID cannot be empty');
    }
    return new UserGameId(value.trim());
  }

  get value(): string {
    return this._value;
  }

  equals(other: UserGameId): boolean {
    return this._value === other._value;
  }
}

File: backend-go/internal/domain/model/user_game_id.go
package model

type UserGameID struct {
	ID
}

func NewUserGameID() UserGameID {
	return UserGameID{ID: NewID()}
}

func ParseUserGameID(value string) (UserGameID, error) {
	id, err := ParseID(value)
	if err != nil {
		return UserGameID{}, err
	}
	return UserGameID{ID: id}, nil
}

func (i UserGameID) Equals(other UserGameID) bool {
	return i.ID.Equals(other.ID)
}
